### ISSUE TITLE
Changefeed status does not forward normally in cyclic replication

### ISSUE THREAD

**Issue by amyangfei** [pingcap/ticdc#679]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

- Setup two TiDB clusters in the A<->A deployment architecture, with multiple tables in upstream and downstream
- Start one cdc server in both TiDB cluster 
- create mark tables
- start changefeed in both cluster

2. What did you expect to see?

changefeed is replicated normally

3. What did you see instead?

- resolved ts and checkpoint ts is nor forwarded
- a lot of following logs

```
[2020/06/21 00:00:31.479 +08:00] [INFO] [etcd.go:110] ["update taskStatus"] [status="{\"tables\":{\"1029\":{\"start-ts\":417508562455756802,\"mark-table-id\":1646},\"1042\":{\"start-ts\":417508562455756802,\"mark-table-id\":1648},\"1107\"
:{\"start-ts\":417508562455756802,\"mark-table-id\":1656},\"1133\":{\"start-ts\":417508562455756802,\"mark-table-id\":1636},\"1193\":{\"start-ts\":417508562455756802,\"mark-table-id\":1638},\"1208\":{\"start-ts\":417508562455756802,\"mark
-table-id\":1652},\"1212\":{\"start-ts\":417508562455756802,\"mark-table-id\":1662},\"1217\":{\"start-ts\":417508562455756802,\"mark-table-id\":1664},\"1255\":{\"start-ts\":417508562455756802,\"mark-table-id\":1660},\"1625\":{\"start-ts\"
:417508562455756802,\"mark-table-id\":1650},\"49\":{\"start-ts\":417508562455756802,\"mark-table-id\":1668},\"53\":{\"start-ts\":417508562455756802,\"mark-table-id\":1654}},\"operation\":null,\"admin-job-type\":0}"]
[2020/06/21 00:00:31.481 +08:00] [INFO] [etcd.go:123] ["outdated table infos, ignore update taskStatus"]
[2020/06/21 00:00:31.848 +08:00] [INFO] [changefeed.go:374] ["dispatch table success"] [captureID=aa7362a6-e00d-46ec-940e-324f2aa94b75] [status="{\"tables\":{\"1029\":{\"start-ts\":417508562455756802,\"mark-table-id\":1646},\"1042\":{\"st
art-ts\":417508562455756802,\"mark-table-id\":1648},\"1107\":{\"start-ts\":417508562455756802,\"mark-table-id\":1656},\"1133\":{\"start-ts\":417508562455756802,\"mark-table-id\":1636},\"1193\":{\"start-ts\":417508562455756802,\"mark-table
-id\":1638},\"1208\":{\"start-ts\":417508562455756802,\"mark-table-id\":1652},\"1212\":{\"start-ts\":417508562455756802,\"mark-table-id\":1662},\"1217\":{\"start-ts\":417508562455756802,\"mark-table-id\":1664},\"1255\":{\"start-ts\":41750
8562455756802,\"mark-table-id\":1660},\"1625\":{\"start-ts\":417508562455756802,\"mark-table-id\":1650},\"49\":{\"start-ts\":417508562455756802,\"mark-table-id\":1668},\"53\":{\"start-ts\":417508562455756802,\"mark-table-id\":1654}},\"ope
ration\":{},\"admin-job-type\":0}"]
[2020/06/21 00:00:32.349 +08:00] [INFO] [changefeed.go:374] ["dispatch table success"] [captureID=aa7362a6-e00d-46ec-940e-324f2aa94b75] [status="{\"tables\":{\"1029\":{\"start-ts\":417508562455756802,\"mark-table-id\":1646},\"1042\":{\"st
art-ts\":417508562455756802,\"mark-table-id\":1648},\"1107\":{\"start-ts\":417508562455756802,\"mark-table-id\":1656},\"1133\":{\"start-ts\":417508562455756802,\"mark-table-id\":1636},\"1193\":{\"start-ts\":417508562455756802,\"mark-table
-id\":1638},\"1208\":{\"start-ts\":417508562455756802,\"mark-table-id\":1652},\"1212\":{\"start-ts\":417508562455756802,\"mark-table-id\":1662},\"1217\":{\"start-ts\":417508562455756802,\"mark-table-id\":1664},\"1255\":{\"start-ts\":41750
8562455756802,\"mark-table-id\":1660},\"1625\":{\"start-ts\":417508562455756802,\"mark-table-id\":1650},\"49\":{\"start-ts\":417508562455756802,\"mark-table-id\":1668},\"53\":{\"start-ts\":417508562455756802,\"mark-table-id\":1654}},\"ope
ration\":{},\"admin-job-type\":0}"]
[2020/06/21 00:00:32.848 +08:00] [INFO] [changefeed.go:374] ["dispatch table success"] [captureID=aa7362a6-e00d-46ec-940e-324f2aa94b75] [status="{\"tables\":{\"1029\":{\"start-ts\":417508562455756802,\"mark-table-id\":1646},\"1042\":{\"st
art-ts\":417508562455756802,\"mark-table-id\":1648},\"1107\":{\"start-ts\":417508562455756802,\"mark-table-id\":1656},\"1133\":{\"start-ts\":417508562455756802,\"mark-table-id\":1636},\"1193\":{\"start-ts\":417508562455756802,\"mark-table
-id\":1638},\"1208\":{\"start-ts\":417508562455756802,\"mark-table-id\":1652},\"1212\":{\"start-ts\":417508562455756802,\"mark-table-id\":1662},\"1217\":{\"start-ts\":417508562455756802,\"mark-table-id\":1664},\"1255\":{\"start-ts\":41750
8562455756802,\"mark-table-id\":1660},\"1625\":{\"start-ts\":417508562455756802,\"mark-table-id\":1650},\"49\":{\"start-ts\":417508562455756802,\"mark-table-id\":1668},\"53\":{\"start-ts\":417508562455756802,\"mark-table-id\":1654}},\"ope
ration\":{},\"admin-job-type\":0}"]
```

4. Versions of the cluster

    - Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

        ```
        master
        ```

    - TiCDC version (execute `cdc version`):

        ```
        master
        ```

**Comment by overvenus** [comment]
Could you show table schemas?

**Comment by amyangfei** [comment]
https://github.com/pingcap/ticdc/blob/master/cmd/client_changefeed.go#L373
should we use `ineligibleTables` instead of `allTables`

**Comment by amyangfei** [comment]
> Could you show table schemas?

exists some ineligible tables.

**Comment by amyangfei** [comment]
Found one similar scenario:
- upstream have several normal tables that can be replicated, create mark tables.
- create a changefeed, replication is normal
- stop cdc server
- create an ineligible table in upstream, such as `create table t (id int);`
- start the cdc server, reload old changefeed
- following logs infinitely

```
[2020/06/22 12:09:28.409 +08:00] [INFO] [changefeed.go:333] ["balance table info delay, wait mark table"] [changefeed=af34c287-54f2-4333-b5be-c8342ebab73e] [tableID=65] [markTableName=repl_mark_test_nt2]
[2020/06/22 12:09:28.413 +08:00] [INFO] [changefeed.go:374] ["dispatch table success"] [captureID=de109146-e891-4380-a27b-11956092b2dd] [status="{\"tables\":{\"45\":{\"start-ts\":417542481192943617,\"mark-table-id\":53},\"47\":{\"start-ts\":417542481192943617,\"mark-table-id\":57},\"71\":{\"start-ts\":417542481192943617,\"mark-table-id\":55}},\"operation\":{},\"admin-job-type\":0}"]
[2020/06/22 12:09:28.825 +08:00] [INFO] [etcd.go:110] ["update taskStatus"] [status="{\"tables\":{\"45\":{\"start-ts\":417542481192943617,\"mark-table-id\":53},\"47\":{\"start-ts\":417542481192943617,\"mark-table-id\":57},\"71\":{\"start-ts\":417542481192943617,\"mark-table-id\":55}},\"operation\":null,\"admin-job-type\":0}"]
[2020/06/22 12:09:28.828 +08:00] [INFO] [processor.go:464] ["update task status"] [status="{\"tables\":{\"45\":{\"start-ts\":417542481192943617,\"mark-table-id\":53},\"47\":{\"start-ts\":417542481192943617,\"mark-table-id\":57},\"71\":{\"start-ts\":417542481192943617,\"mark-table-id\":55}},\"operation\":null,\"admin-job-type\":0}"] [position="{\"checkpoint-ts\":417542481192943617,\"resolved-ts\":417542692611817473,\"count\":0}"]
[2020/06/22 12:09:28.829 +08:00] [INFO] [changefeed.go:333] ["balance table info delay, wait mark table"] [changefeed=af34c287-54f2-4333-b5be-c8342ebab73e] [tableID=65] [markTableName=repl_mark_test_nt2]
[2020/06/22 12:09:28.832 +08:00] [INFO] [changefeed.go:374] ["dispatch table success"] [captureID=de109146-e891-4380-a27b-11956092b2dd] [status="{\"tables\":{\"45\":{\"start-ts\":417542481192943617,\"mark-table-id\":53},\"47\":{\"start-ts\":417542481192943617,\"mark-table-id\":57},\"71\":{\"start-ts\":417542481192943617,\"mark-table-id\":55}},\"operation\":{},\"admin-job-type\":0}"]
[2020/06/22 12:09:28.909 +08:00] [INFO] [changefeed.go:333] ["balance table info delay, wait mark table"] [changefeed=af34c287-54f2-4333-b5be-c8342ebab73e] [tableID=65] [markTableName=repl_mark_test_nt2]
[2020/06/22 12:09:28.911 +08:00] [INFO] [changefeed.go:374] ["dispatch table success"] [captureID=de109146-e891-4380-a27b-11956092b2dd] [status="{\"tables\":{\"45\":{\"start-ts\":417542481192943617,\"mark-table-id\":53},\"47\":{\"start-ts\":417542481192943617,\"mark-table-id\":57},\"71\":{\"start-ts\":417542481192943617,\"mark-table-id\":55}},\"operation\":{},\"admin-job-type\":0}"]
```

**Comment by amyangfei** [comment]
detail logs: https://transfer.sh/iEiEl/cdc_20200621.log.tar.gz
ddl history: https://transfer.sh/TzpZY/ddl_history_20200621.log.tar.gz
